### PR TITLE
docs: update experimental inline notification content for tags

### LINKED ISSUE #4079: [Tag] revise experimental inline notification content
## Acceptance criteria

Revise the experimental inline notification on the Tag Usage tab page to be more clear.

We need to get this message across:
- [x] Dimissible tags have been separated into its own variant. It is experimental.
- [x] The filter prop will be replaced by the Dismissible tag variant in v12.
- [x] Selectable and Operational tags are new variants. They are experimental.

Screenshot 2024-05-23 at 9 23 55 AM

### PR DESCRIPTION
**Pull request by laurenmrice**
Closes #4079 

Updates the experimental inline notification for the Tag component on the Usage tab to be less confusing and note that the read-only tag variant and filter prop will not be deprecated and remain stable but to use the dismissible one moving forward for dismissible functionality. 

*This notification can get pretty long. If anyone else has suggestions on how to word this and shorten it please do! Thanks.

---

**Changed**

- Changed content of the inline notification on the Tag Usage tab.

### CODE REVIEW

File: src/pages/components/tag/usage.mdx [hunk 1]
```diff
@@ -17,10 +17,7 @@ them.
 
 <InlineNotification>
 
-[Experimental tags](https://react.carbondesignsystem.com/?path=/docs/experimental-unstable-interactivetag--overview)
-are now available. Though we are not deprecating the current tags, we encourage
-all design teams to use the new dismissible, operational, and selectable tag
-variants in their products moving forward.
+[Experimental tags](https://react.carbondesignsystem.com/?path=/docs/experimental-unstable-interactivetag--overview)for dismissible, selectable, and operational functionality are now available. Read-only tags and the filter prop remain stable in the system. Although we are not deprecating the filter prop in the read-only tag, we encourage all design teams to use the experimental "dismissible" tag in their products moving forward and using the other experimental tags when applicable.
```
Comment (laurenmrice) [inline src/pages/components/tag/usage.mdx:20]:
```suggestion
[Experimental tags](https://react.carbondesignsystem.com/?path=/docs/experimental-unstable-interactivetag--overview) for dismissible, selectable, and operational functionality are now available. Read-only tags and the filter prop remain stable in the system. Although we are not deprecating the filter prop in the read-only tag, we encourage all design teams to use the experimental "dismissible" tag in their products moving forward.
```

**Review by alisonjoseph — APPROVED** [review]

**Review by Kritvi-bhatia17 — COMMENTED** [review]
Although it's not a significant change, I thought of removing a few words to shorten it. Feel free to choose whichever version you prefer.

> Experimental tags for dismissible, selectable, and operational functionality are now available.
> 
> Read-only tags and the filter prop remain stable. While we won't deprecate the filter prop, we encourage design teams to use the experimental 'dismissible' tag moving forward.

**Review by aagonzales — COMMENTED** [review]
@laurenmrice Its still a little confusing to me. This message is both announcing new variants and also offering a recommended alternative to an old prop?

Maybe something more direct like this:
> [Experimental tags](https://react.carbondesignsystem.com/?path=/docs/experimental-unstable-interactivetag--overview&_ga=2.239566336.261006391.1717082399-1401223795.1717082399&_gl=1*dn08ye*_ga*MTQwMTIyMzc5NS4xNzE3MDgyMzk5*_ga_FYECCCS21D*MTcxNzA4MjM5OS4xLjEuMTcxNzA4MjQxMC4wLjAuMA..) for dismissible, selectable, and operational functionality are now available as additional tag variants. Dismissible tag is being offered as a recommended alternative solution to the filter prop for x reason.

**Comment by laurenmrice** [comment]
@Kritvi-bhatia @aagonzales Thank you so much for looking at this, I have been struggling with it a bit.


Anna with your last rewrite to make it more direct, what do you think about adding this to the end? That is essentially what will end up happening:

> [Experimental tags](https://react.carbondesignsystem.com/?path=/docs/experimental-unstable-interactivetag--overview&_ga=2.239566336.261006391.1717082399-1401223795.1717082399&_gl=1*dn08ye*_ga*MTQwMTIyMzc5NS4xNzE3MDgyMzk5*_ga_FYECCCS21D*MTcxNzA4MjM5OS4xLjEuMTcxNzA4MjQxMC4wLjAuMA..) for dismissible, selectable, and operational functionality are now available as additional tag variants. Dismissible tag is the recommended alternative solution to the filter prop and will eventually replace the prop once v12 is available.

**Review by laurenmrice — COMMENTED** [review]
updated content

**Review by aagonzales — APPROVED** [review]

**Pull request merged**